src/ol/layer/Graticule.js

```diff
@@ -484,9 +483,11 @@ class Graticule extends VectorLayer {
     // extents may be passed in different worlds, to avoid endless loop we use only one
     const realExtent = extent.slice();
     if (this.projection_ && this.getSource().getWrapX()) {
-      wrapX(realExtent, this.projection_);
+      wrapExtentX(realExtent, this.projection_);
     }
-    if (this.loadedExtent_ && !containsExtent(bufferExtent(this.loadedExtent_, resolution / 2), realExtent)) {
+    realExtent[0] = Math.round(realExtent[0] * 1e8) / 1e8;
+    realExtent[2] = Math.round(realExtent[2] * 1e8) / 1e8;
+    if (this.loadedExtent_ && !equals(this.loadedExtent_, realExtent)) {
```

Comment:
```suggestion
    if (this.loadedExtent_ && !equals(this.loadedExtent_, realExtent.map(function(n) { return Math.round(n * resolution) / resolution; })) {
```
In combination with the above suggestion, this would round the extent to pixels, which should be accurate enough and would avoid the arbitrary `1e8` figure.

Reply:
I also tried 

    if (this.loadedExtent_ && (
      !containsExtent(bufferExtent(this.loadedExtent_, resolution / 2), realExtent) ||
      !containsExtent(bufferExtent(realExtent, resolution / 2), this.loadedExtent_)
    )) {


but that fails when resolutions become so small relative to large coordinate values that buffering has no effect.

Any rounding has the potential for a different last digit.  A more general approach which would avoid that and might be useful elsewhere would be a near equality method similar to that used in the tests which checks values are within JavasScript's approx 15 digit accuracy e.g.

`Math.abs(a - b) <= Math.abs((a + b) * 1e-15)`

Reply:
What do you use to test this pull request? I.e. where do you check for the infinite loops you mentioned several times? Which example or test case?

Reply:
I set up a sandbox https://codesandbox.io/s/simple-11f0s patched with the changes and logging the extents used in the strategy function to debug the looping.  The opening at +180 problem is easily reproduced by changing `>=` to `>`  The rounding problems occur intermittently if the rounding is commented out.

Reply:
If you could turn that into a unit or rendering test, it would be great.